DownFall/Rendering/Renderer.swift

```diff
@@ -793,3 +797,12 @@ extension Renderer: SettingsDelegate {
         InputQueue.append(Input(.pause))
     }
 }
+
+
+// MARK: Saving Game State
+extension Renderer {
+    public func goalTrackingData() -> [GoalTracking]
+    {
+        return levelGoalTracker.goalProgress
+    }
+}
```

Comment:
This kinda sucks and break encapsulation.  Is there a better way around this?

Reply:
Otherwise we need to heavily comment

Reply:
Fixed it by moving levelgoaltracker as a injected dependency for renderer.  Now GameScene holds references to everything that needs to be saved